rocAL/rocAL/include/augmentations/geometry_augmentations/node_resize.h

```diff
@@ -32,9 +34,18 @@ class ResizeNode : public Node
     unsigned int get_dst_height() { return _outputs[0]->info().height_single(); }
     vx_array get_src_width() { return _src_roi_width; }
     vx_array get_src_height() { return _src_roi_height; }
+    void init(unsigned dest_width, unsigned dest_height, RocalResizeScalingMode scaling_mode,
+              std::vector<unsigned> max_size, RocalResizeInterpolationType interpolation_type);
+    void adjust_out_roi_size();
 protected:
     void create_node() override;
     void update_node() override;
 private:
-    vx_array  _dst_roi_width , _dst_roi_height ;
+    vx_array  _dst_roi_width , _dst_roi_height;
+    unsigned _dest_width, _dest_height;
+    int _interpolation_type;
+    RocalResizeScalingMode _scaling_mode;
+    std::vector<float> _src_roi_size;
```

Comment:
Why src_roi_size is in float?

Reply:
We require the src size values to be retained as float, inorder to avoid precision loss which may lead to a +1 or -1 value while rounding.